Add new noSelectStar analyzer

lieut-data opened the pull request:
#### Summary
This guards against forwards compatibility issues with `SELECT *`.

I won't actually enable this in the monorepo until https://mattermost.atlassian.net/browse/MM-62110 is resolved.

#### Ticket Link
Relates-to: https://mattermost.atlassian.net/browse/MM-62169

lieut-data committed:
Add new noSelectStar analyzer

This guards against forwards compatibility issues from adding new
columns an old server won't know how to handle.

Relates-to: https://mattermost.atlassian.net/browse/MM-62169

jespino reviewed (approved):
LGTM

agnivade commented on noSelectStar/noSelectStar.go:
```diff
@@ -0,0 +1,57 @@
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspect := func(node ast.Node) bool {
```
Can we declare these as external functions? It looks more readable that way. Also, would `inspectStrings` be a better name here?

lieut-data replied:
I've simplified this to just a single `inspect` -- this matches the pattern elsewhere in this repo, and also avoids having to keep `pass` in scope for the callback.

agnivade commented on noSelectStar/testdata/src/a/a.go:
```diff
@@ -0,0 +1,37 @@
+	Select("id", "name", "*") // want `do not use Select with \*: explicitly select the needed columns instead`
+
+	// These should not trigger warnings for Select function
+	Select("id, name")
```
Can we also add a test case like:

```
query := s.getQueryBuilder().
		Select("*").
		From("Channels").
		Where(sq.Eq{"Id": ids})
```

to test for completeness?

lieut-data replied:
Good catch: the previous code was entirely incompatible with our actual pattern here. I confess to offloading too much to aider here: I suppose this is analogous to the self-driving car problem where the drivers risk not paying sufficient attention to take over when things don't go right.

I still used aider to fix this, but enhanced the test cases manually.

agnivade replied:
Good that we have humans to review code :joy:

lieut-data committed:
test: Add squirrel builder pattern test case for SELECT *

The commit adds a test case for the squirrel builder pattern to verify that the linter catches SELECT * usage even when using method chaining. Specifically:

1. Imported the squirrel package as `sq`
2. Added a test case using `sq.StatementBuilder.PlaceholderFormat(sq.Dollar).Select("*")...`
3. Added an appropriate `want` comment to verify the linter catches the SELECT * usage

This ensures the linter works correctly with fluent SQL builder libraries like squirrel.

lieut-data committed:
docs: Add noSelectStar analyzer to README documentation
